chiselFrontend/src/main/scala/chisel3/core/MonoConnect.scala

```diff
@@ -107,14 +109,21 @@ object MonoConnect {
             case MonoConnectException(message) => throw MonoConnectException(s".$field$message")
           }
         }
-
+      // Source is DontCare - it may be connected to anything. It generates a defInvalid for the sink.
+      case (sink, DontCare) => pushCommand(DefInvalid(sourceInfo, sink.lref))
```

Comment:
Is this being duplicated with the logic in issueConnect? Abstraction level wise this seems out of place with the rest of the cases here...

Reply:
No, it deals with the case where we connect DontCares to Vecs or Bundles. It eliminates the logic that had to pull those aggregates apart to get down to their leaves. This is closer to the way the current code (non-invalidateapi) deals with "is invalid".

Reply:
Do we want to pull apart vecs and bundles as with the rest of the code? for example, what happens if you invalidate a bidirectional bundle?

Reply:
Good question. If we do pull apart aggregates (for the purpose of DontCare connections), firrtl could be more helpful reporting which elements of an aggregate aren't completely initialized. I think this is an implementation issue which could be resolved post release - it shouldn't change the API.